builder.go

```diff
@@ -105,18 +109,27 @@ func (eb ErrorBuilder) Create() *Error {
 type callStackBuildMode int
 
 const (
-	stackTraceCollect callStackBuildMode = 1
-	stackTraceBorrow  callStackBuildMode = 2
-	stackTraceEnhance callStackBuildMode = 3
-	stackTraceOmit    callStackBuildMode = 4
+	stackTraceCollect         callStackBuildMode = 1
+	stackTraceBorrowOrCollect callStackBuildMode = 2
+	stackTraceBorrowOnly      callStackBuildMode = 3
+	stackTraceEnhance         callStackBuildMode = 4
+	stackTraceOmit            callStackBuildMode = 5
 )
 
 func (eb ErrorBuilder) assembleStackTrace() *stackTrace {
 	switch eb.mode {
 	case stackTraceCollect:
 		return eb.collectOriginalStackTrace()
-	case stackTraceBorrow:
-		return eb.borrowStackTraceFromCause()
+	case stackTraceBorrowOnly, stackTraceBorrowOrCollect:
```

Comment:
Very minor: perhaps making this into 2 separate cases with a little copy-paste but without another `if eb.mode ==` would improve readability.

Reply:
You're right.